### ISSUE TITLE
💅 useArrowFunction breaks working code on safe lint

### ISSUE THREAD

**Issue by RSharmaBB** [biomejs/biome#4967]
### Environment information

```bash
Repo to reproduce: https://github.com/RSharmaBB/biome-arrow-reproduce

It looks like the `useArrowFunction` rule is not adding parenthesis when needed resulting in breaking changes when running `biome check --write`. 

This is expected to be a safe action according to the docs
https://biomejs.dev/guides/getting-started/
```

### Rule name

useArrowFunction

### Playground link

https://github.com/RSharmaBB/biome-arrow-reproduce

### Expected result

It should have wrapped the returned object in parenthesis.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
We can preview lint fixes in our playground now, see the "Analyzer Fixes" tab: https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAbwBiAGoAIAA9ACAAewAKACAAIABtAGUAcwBzAGEAZwBlADoAIABmAHUAbgBjAHQAaQBvAG4AIAAoAGEAcgBnACkAIAB7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAHsACgAgACAAIAAgACAAIABmAG8AbwA6ACAAIgBtAGUAcwBzAGEAZwBlACAAcgBlAHMAcABvAG4AcwBlACAAdABvACAAZgBvAG8AIgAsAAoAIAAgACAAIAAgACAAYgBhAHIAOgAgACIAbQBlAHMAcwBhAGcAZQAgAHIAZQBzAHAAbwBuAHMAZQAgAHQAbwAgAGIAYQByACIALAAKACAAIAAgACAAIAAgAGQAdQBjAGsAOgAgACIAbQBlAHMAcwBhAGcAZQAgAHIAZQBzAHAAbwBuAHMAZQAgAHQAbwAgAGQAdQBjAGsAIgAsAAoAIAAgACAAIAB9AFsAYQByAGcAXQAKACAAIAB9AAoAfQA7AA%3D%3D

I can confirm that the fixed code does not parse and execute, and parenthesis should be added around the function body to make it work.
